src/Check/D7/EntityReferenceAutocomplete.php

```diff
@@ -0,0 +1,176 @@
+<?php
+/**
+ * @file
+ * Contains SiteAudit\Check\D7\EntityReferenceAutocomplete
+ */
+
+namespace SiteAudit\Check\D7;
+
+use SiteAudit\Check\Check;
+use SiteAudit\AuditResponse\AuditResponse;
+use SiteAudit\Annotation\CheckInfo;
+
+/**
+ * @CheckInfo(
+ *  title = "Entity reference autocomplete",
+ *  description = "Ensure that entity reference fields are configured correctly.",
+ *  remediation = "Change the following field definitions to autocomplete - <br/>:error_replace.",
+ *  success = "Found <code>:num_fields</code> entity reference field:num_plural configured correctly.",
+ *  failure = "Found configuration errors in <code>:error_count</code> entity reference field:error_plural.",
+ *  exception = "Could not find any entity reference fields.",
+ *  not_available = "No entity reference fields.",
+ * )
+ */
+class EntityReferenceAutocomplete extends Check {
+
+  protected $errors = [];
+
+  /**
+   * Identify if entity reference fields are displaying select lists.
+   *
+   * Sites can see crippling performance if an entity reference field is being
+   * used to display entities from a large node pool. This process will never
+   * fail with max_execution_time and as a result can cause PHP-FPM to backup
+   * and queue requests while it deals with an errant entity reference field.
+   *
+   * @return int AuditResponse::AUDIT_SUCCESS or similar.
+   */
+  protected function check() {
+    $valid = 0;
+
+    try {
+      $output = $this->context->drush->sqlQuery('SELECT fc.field_name, fc.data, fci.data FROM {field_config} fc JOIN {field_config_instance} fci ON fc.id = fci.field_id WHERE fc.type = \'entityreference\'');
```

Comment:
Can we do double quotes on the outside, and unescaped single on the inside. I hope all checks follow this standard (the quoting escaping is very complex).